### PR TITLE
Changes the link to download Kafka

### PR DESCRIPTION
**Pull request by davinci26**
Signed-off-by: Sotiris Nanopoulos <sonanopo@microsoft.com>
Commit Message:

Updates link for `kafka_server_binary`

Additional Description: N/A
Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

**Comment by davinci26** [comment]
@envoyproxy/dependency-shepherds @snowp @wrowe

**Comment by moderation** [comment]
The old version we are on `2.4.1` has aged out of the mirrors with `2.6.1` being the oldest version on the approved mirror - https://mirrors.gigenet.com/apache/kafka/ /cc @adamkotwasinski 

/lgtm deps

**Comment by yanavlasov** [comment]
@davinci26 format is failing on python changes.

File: bazel/repository_locations.bzl [hunk 1]
```diff
@@ -802,9 +802,9 @@ REPOSITORY_LOCATIONS_SPEC = dict(
         project_desc = "Open-source distributed event streaming platform",
         project_url = "https://kafka.apache.org",
         version = "2.4.1",
-        sha256 = "2177cbd14118999e1d76fec628ca78ace7e6f841219dbc6035027c796bbe1a2a",
-        strip_prefix = "kafka_2.12-{version}",
-        urls = ["https://mirrors.gigenet.com/apache/kafka/{version}/kafka_2.12-{version}.tgz"],
+        sha256 = "d1064e87c2f1aee64f659ca9fd9d491d6d35b37e3bdb1fbd588d6100111e5e49",
+        strip_prefix = "kafka-{version}-src",
+        urls = ["https://archive.apache.org/dist/kafka/{version}/kafka-{version}-src.tgz"],
```
Comment (adamkotwasinski) [inline bazel/repository_locations.bzl:807]: isn't that the change from compiled binary to zip with sources?
could you please try to run
`bazel test //test/extensions/filters/network/kafka/broker/integration_test:kafka_broker_integration_test`

Comment (davinci26) [reply bazel/repository_locations.bzl:807]: Makes sense, I think you are right. Changing it to the binaries

**Comment by davinci26** [comment]
The kafka integration test is not running on my dev box but since the test is marked as manual, I am not sure if it is supposed to run on Windows. Pushing on the CI but since the executable is the same I expect it should work

**Comment by dio** [comment]
Let me retest again. 

/retest

**Review by yanavlasov — APPROVED** [review]

**Comment by davinci26** [comment]
The error in the verify examples is:

```
Successfully built 7dccf19c536a
Successfully tagged zipkin-tracing_service2:latest
Pulling zipkin (openzipkin/zipkin:)...
latest: Pulling from openzipkin/zipkin
write /var/lib/docker/tmp/GetImageBlob382198399: no space left on device
ERROR: starting zipkin .
```

cc @dio

**Review by lizan — APPROVED** [review]

**Pull request merged**